### ISSUE TITLE
Record package names during coverage capture

### ISSUE THREAD

**Issue by arcivanov** [coveragepy/coveragepy#936]
Coverage object accepts both source paths and package names.
However, the currently captured schema does not record package names even when the source is specified in the form of the package names - only traced files are recorded.

It would be very helpful for reconciling coverage from multiple identical files located in different (unknown) locations on the file system, if Analysis object could provide `package_name` in addition to `filename`.

**Comment by arcivanov** [comment]
In my work on PyBuilder 0.12 a lot of awkwardness related to how Coverage captures and stores the coverage info has been becoming increasingly apparent.

Imaging the following scenario:

Many tests are run on the identical code from the different environments in different locations. For example.

1. `~/project_root/src/main/python/*`
2. `~/project_root/target/dist/project_name.0.1.2.dev/*`
3. `/tmp/project_root_803ydfs843i/src/main/python/*`

Basically, I think what coverage does with `coverage.files.RELATIVE_DIR` and other `PathAlias` magic are improper half-measures. There is no way to determine reliably what the package name is from the files alone.

Here's the proposed solution:

1. Every `__module__` has a `__file__` therefore the location of the file for a module/package can be determined easily.
2. Both package/module name and the file are recorded by the tracer.

**Comment by arcivanov** [comment]
Yet another way to look at it is as follows:

For every module loaded one can find the file path easily. However, to understand what *module* that *file* belongs too (or what *relative path* for that file is), i.e. the reverse process, can **only** be done at runtime. Module loading is the result of inspecting `sys.path`, `sys.meta_path`, which determine what the package/module names is relative to currently configured directories/loader settings etc.

I.e. the song and dance with `files.RELATIVE_DIR` cannot help - if the `sys.path`, `current directory` or `sys.meta_path` change at runtime, coverage will record correct absolute files but will derive completely bogus relative files/package names.

Therefore, even recording `sys.path` and `cwd` at runtime for the whole coverage run is not sufficient to determine module/package names correctly as those can change during coverage from file to file.

Thus, the only reliable way to determine what package name(*s*) have corresponded to which file when the file was traced is to record that information during tracing. All other cases eventually will produce bogus results.
